### PR TITLE
Implement configuration stuff

### PR DESCRIPTION
**Pull request by kinggoesgaming**
Related MinigameCore/MinigameCoreAPI#21

### CODE REVIEW

File: src/main/java/io/github/minigamecore/plugin/config/Configurations.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import io.github.minigamecore.api.util.config.Configuration;
+import io.github.minigamecore.plugin.MinigameCore;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Util class for handling all the Minigame Core specific configurations.
```
Comment (Flibio) [inline src/main/java/io/github/minigamecore/plugin/config/Configurations.java]: No space in MinigameCore.

File: src/main/java/io/github/minigamecore/plugin/MinigameCore.java [hunk 2]
```diff
@@ -65,10 +73,37 @@ private MinigameCore(@ConfigDir(sharedRoot = false) Path configDir, Injector def
...
+
+    @Listener(order = LATE)
+    public void onStoppingLate(final GameStoppingEvent event) {
+        ((ConfigurationManagerImpl) getServiceManager().provideUnchecked(MinigameService.class).getConfigurationManager()).saveAllConfigurations();
+    }
+
+    // TODO move this to another class later
+    @Listener
+    public void onReload(final GameReloadEvent event) {
+        ((ConfigurationManagerImpl) defaultInjector.getInstance(ConfigurationManager.class)).saveAllConfigurations();
```
Comment (Flibio) [inline src/main/java/io/github/minigamecore/plugin/MinigameCore.java]: Shouldn't the configurations be loaded instead of being saved?

Comment (kinggoesgaming) [inline src/main/java/io/github/minigamecore/plugin/MinigameCore.java]: Copy pasta woops

File: src/main/java/io/github/minigamecore/plugin/MinigameCore.java [hunk 3]
```diff
@@ -65,10 +73,37 @@ private MinigameCore(@ConfigDir(sharedRoot = false) Path configDir, Injector def
...
         getServiceManager().setProvider(this, MinigameService.class, defaultInjector.getInstance(MinigameService.class));
+
+        getAll().forEach((configuration -> defaultInjector.getInstance(ConfigurationManager.class).register(this, configuration)));
+
+        ((ConfigurationManagerImpl) defaultInjector.getInstance(ConfigurationManager.class)).loadAllConfigurations();
+    }
+
+    @Listener(order = LATE)
+    public void onPreInitLate(final GamePreInitializationEvent event) {
+        ((ConfigurationManagerImpl) defaultInjector.getInstance(ConfigurationManager.class)).saveAllConfigurations();
```
Comment (Flibio) [inline src/main/java/io/github/minigamecore/plugin/MinigameCore.java]: Should this be loading too, or is it save?

Comment (kinggoesgaming) [inline src/main/java/io/github/minigamecore/plugin/MinigameCore.java]: This is saving... Merging assets and then saving

**Comment by Flibio** [comment]
Squash when you are ready.

**Comment by kinggoesgaming** [comment]
Ready :)

**Pull request merged**